dj_hetmech/models.py

```diff
@@ -0,0 +1,69 @@
+"""
+Database schema.
+
+See https://django-extensions.readthedocs.io/en/latest/graph_models.html for
+exporting a schema visualization. Must install django-extensions and pygraphviz / pydotplus,
+then run:
+
+```
+python manage.py graph_models --output=media/models-schema.svg dj_hetmech
+python manage.py graph_models --output=media/models-schema.png dj_hetmech
+```
+
+References:
+  https://docs.djangoproject.com/en/2.1/ref/models/fields/
+  https://docs.djangoproject.com/en/2.1/ref/models/options/
+"""
+
+from django.db import models
+
+
+class Metanode(models.Model):
+    identifier = models.CharField(primary_key=True, max_length=50)
+    abbreviation = models.CharField(max_length=10)
+
+
+class Node(models.Model):
+    metanode = models.ForeignKey(to='Metanode', on_delete=models.PROTECT)
+    identifier = models.CharField(max_length=50)
+    identifier_type = models.CharField(max_length=50, choices=[
+        ('str', 'string'),
+        ('int', 'integer'),
+    ])
+    name = models.CharField(max_length=200)
+    url = models.URLField(blank=True)
+
+    class Meta:
+        unique_together = ('metanode', 'identifier')
+
+
+class Metapath(models.Model):
+    abbreviation = models.CharField(primary_key=True, max_length=20)
+    verbose = models.CharField(max_length=200)
+    verbose_pretty = models.CharField(max_length=200)
+    source = models.ForeignKey(to='Metanode', on_delete=models.PROTECT, related_name='metapath_source')
+    target = models.ForeignKey(to='Metanode', on_delete=models.PROTECT, related_name='metapath_target')
```

Comment:
Without `related_name`, I was getting the following errors:

```
dj_hetmech.Metapath.source: (fields.E304) Reverse accessor for 'Metapath.source' clashes with reverse accessor for 'Metapath.target'.
	HINT: Add or change a related_name argument to the definition for 'Metapath.source' or 'Metapath.target'.
dj_hetmech.Metapath.target: (fields.E304) Reverse accessor for 'Metapath.target' clashes with reverse accessor for 'Metapath.source'.
	HINT: Add or change a related_name argument to the definition for 'Metapath.target' or 'Metapath.source'.
```

Seems like we could also set `related_name='+'` to suppress backwards relations.
https://docs.djangoproject.com/en/2.1/ref/models/fields/#django.db.models.ForeignKey.related_name

However, I don't fully understand backwards relations and when we would use them.